sdk/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -220,24 +220,42 @@ type PermissionClaim struct {
 	IdentityHash string `json:"identityHash,omitempty"`
 }
 
-// +kubebuilder:validation:XValidation:rule="has(self.__namespace__) || has(self.name)",message="at least one field must be set"
+const (
+	ResourceSelectorAll = "*"
+)
+
+// ResourceSelectorName validates a string for use with a PermissionClaim.
+//
+// +kubebuilder:validation:Type=string
+// +kubebuilder:validation:Pattern=`^(\*|[a-z][-a-z0-9]*[a-z0-9])$`
```

Comment:
not all resources have that restricted names afaik (CRDs do). E.g. RBAC objects can have `:`.

Reply:
Maybe let's just do minLength=1?